crates/solidity/testing/utils/src/version_pragmas/mod.rs

```diff
@@ -20,19 +20,26 @@ pub fn extract_version_pragmas(
     let language = &Language::new(latest_version.to_owned())?;
     let output = language.parse(ProductionKind::SourceUnit, source);
 
-    let mut cursor = output.create_tree_cursor();
     let mut pragmas = vec![];
-    while let Some(rule_node) = cursor.find_rule_with_kind(&[RuleKind::VersionPragmaExpression]) {
```

Comment:
Unfortunately this is recursive, which for something like `X || Y` would match all three of `X || Y`, `X`, and `Y`. We need to skip children when found.

Reply:
Since this is DFS pre-order, I imagine the parent should be matched first and in the loop body we already call `cursor.go_to_next_non_descendent()`, which should skip the subtree elements (i.e. individual versions) in a subsequent `while find_rule_with_kind` loop call. Otherwise, how the previous code handled it? Or was it wrong/matched everything recursively?